stock_quant_package_product_packaging/models/stock_quant_package.py

```diff
@@ -43,12 +56,46 @@ def auto_assign_packaging(self):
                 and pack.single_product_id
                 and pack.single_product_qty
             ):
-                packaging = self.env["product.packaging"].search(
-                    [
-                        ("product_id", "=", pack.single_product_id.id),
-                        ("qty", "=", pack.single_product_qty),
-                    ],
-                    limit=1,
-                )
-                if packaging:
-                    pack.write({"product_packaging_id": packaging.id})
+                pack._assign_packaging(pack.single_product_id, pack.single_product_qty)
+
+    def _assign_packaging(self, product, quantity):
+        self.ensure_one()
+        packaging = product._find_best_packaging(quantity)
+        if packaging and packaging.qty == quantity:
+            # the call to write will trigger a call to _sync_package_type_from_packaging
+            self.product_packaging_id = packaging
+        elif self.product_packaging_id:
+            self.product_packaging_id = False
+        if packaging:
+            self._sync_package_type_from_packaging(packaging)
+        else:
+            self._sync_package_type_from_single_product(product, quantity)
+
+        if not self.package_type_id and product.package_type_id:
+            self.package_type_id = product.package_type_id
+
+    def _sync_package_type_from_packaging(self, packaging):
```

Comment:
This declaration clashes with this one https://github.com/OCA/stock-logistics-putaway/blob/1043b255ccc94cc137d8d3e4360601daae53527b/stock_storage_type/models/stock_quant_package.py#L70
(same function name, different parameter list) If that other module is installed

Reply:
https://github.com/OCA/stock-logistics-putaway/pull/11/files

Reply:
oh..ok